### DISCUSSION TITLE
How to insert literal "{{string}}" in a template file?

### DISCUSSION THREAD

**Discussion by unikitty37** [twpayne/chezmoi#3870] [Q&A] [answered]
I'm trying to add a config file for lazygit, which also uses the Go template format.

I want to insert this:

```
os:
  edit: "code -n {{filename}}"
```

if `.chezmoi.os` is equal to "darwin".

```
{{ if eq .chezmoi.os "darwin" -}}
os:
  edit: "code -n {{filename}}"
{{ end-}}
```

fails, obviously. But escaping the `{{` and `}}` with backslashes doesn't work — I get "bad character U+005C '\'".

The [Go text template docs](https://pkg.go.dev/text/template) don't say how to do it. I've checked StackOverflow and the [solutions I've found there](https://stackoverflow.com/questions/17641887/how-do-i-escape-and-delimiters-in-go-templates) don't work either.

Please could the solution be added to the documentation? chezmoi isn't the only program to use Go templates, and not everyone is a Go developer :)

**Comment by twpayne** [comment, marked as answer]
Use

    {{ "{{ filename }}" }}

**Reply by unikitty37** [reply]
Thanks! I thought I'd tried that, but I think I must have messed at least one of them up 😊

**Reply by halostatue** [reply]
You can also use [alternate delimiters](https://www.chezmoi.io/reference/templates/directives/#delimiters).

**Reply by unikitty37** [reply]
Ah, that lets me use « and » in that file, which looks a lot cleaner (though I wouldn't do that if other people are going to be editing it, of course, as it's not the easiest to type on a lot of keyboard layouts :)

**Comment by VyvaHart** [comment]


**Reply by twpayne** [reply]
Neither of this approaches works. Did you use ChatGPT to generate your post?
